### PR TITLE
Unify and extend handling of Neuropixels SYNC channel

### LINKED ISSUE #1321: Correct handling of Sync channel for SpikeGLX and Open Ephys
**Describe the bug**
Both SpikeGLX and Open Ephys - Neuropix-PXI can **optionally** save a 385th SYNC channel in the binary file.

While both IOs have a `load_sync_channel` option, all cases are not currently handled. In particular:
- SpikeGLX does not check wheter a SYNC channel is present, and just remove the last channel if `load_sync_channel=False` (see [here](https://github.com/NeuralEnsemble/python-neo/blob/master/neo/rawio/spikeglxrawio.py#L127-L128)).
- Open Ephys handles it a bit betterr, BUT it doesn't throw an error if `load_sync_channel=True`, but not present in the data (same for SpikeGLX)

**Expected behaviour**
SpikeGLX and OpenEphys correctly handle the SYNC option of NP probes in a similar way, handling all possible cases.

**Additional context**
Currently on GIN we only have:
- SpikeGLX with SYNC
- OpenEphys without SYNC

We need to gather:
- SpikeGLX without SYNC (any volunteer? @DradeAW ? )
- Open Ephys with SYNC (I can get one)

### PR DESCRIPTION
**Pull request by alejoe91**
Fixes #1321 

- Handles all options and combinations of SYNC channels in SpikeGLX and Open Ephys.
- Memmaps are sliced in the `_get_analogsignal_chunk` function, to be prepared for #1265

### TODO

- [x] Add test files on GIN
  - [x] SpikeGLX without SYNC
  - [x] SpikeGLX with less than 384 channels
  - [x] Open Ephys with SYNC
- [x]  Extend tests

### CODE REVIEW

**Comment by alejoe91** [comment]
@JuliaSprenger @samuelgarcia extended tests for SpikeGLX and Open Ephys using the newly added GIN files (https://gin.g-node.org/NeuralEnsemble/ephy_testing_data/pulls/110)

Ready to review!

**Comment by alejoe91** [comment]
@samuelgarcia can you take a look at this? This is quite important on our side!

File: neo/rawio/openephysbinaryrawio.py [hunk 1]
```diff
@@ -359,7 +357,15 @@ def _get_signal_t_start(self, block_index, seg_index, stream_index):
 
     def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
                                 stream_index, channel_indexes):
+        info = self._sig_streams[block_index][seg_index][stream_index]
         sigs = self._sig_streams[block_index][seg_index][stream_index]['memmap']
+
+        # take care of SYNC channel
+        channel_names = [ch["channel_name"] for ch in info["channels"]]
+        has_sync_channel = any(["SYNC" in ch for ch in channel_names])
```
Comment (samuelgarcia) [inline neo/rawio/openephysbinaryrawio.py:365]: This should not be done at every chunk bu in the parse header once for all.

**Comment by samuelgarcia** [comment]
Merci.
I made on comments.
As a general feedback, variable name change make the PR very tedious to read and prevent focusing on the real change!

File: neo/rawio/openephysbinaryrawio.py (hunk 1 above)
Comment (alejoe91) [reply neo/rawio/openephysbinaryrawio.py:365]: ok, I'll add a `has_sync_channel` in the `self._sig_streams[block_index][seg_index][stream_index]` dictionary

**Comment by alejoe91** [comment]
@samuelgarcia tried to fix failing Maxwell tests, but that doesn't work. 

On the spikeinterface side, setting an ENV variable in the GH action seems to work: https://github.com/SpikeInterface/spikeinterface/pull/2161

Should I push something similar here?

Nevertheless, it's unrelated to this PR

**Review by samuelgarcia — APPROVED** [review]

**Comment by samuelgarcia** [comment]
Amazing work.

**Pull request merged**